dev-tools/smoke.js

```diff
@@ -0,0 +1,433 @@
+/**
+ * Battle Simulator smoke testing tooling.
+ *
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * @license MIT
+ */
+
+'use strict';
+
+const child_process = require('child_process');
+const shell = cmd => child_process.execSync(cmd, {stdio: 'inherit', cwd: __dirname});
+
+// Run the build script if we're being called from the command line.
```

Comment:
This would probably benefit from documentation for calling it from the command line.

Reply:
Unlike the harness (which will get even more args once I add benchmarking support...), this one is designed to pretty much just run without args. 5f539a9 adds an example. The negative cycles option is kind of hidden on purpose, and passing in format + seed is for debugging purposes (and the error message will tell you about them when relevant).

Reply:
I think it'd benefit from a short summary of what a smoke test is (I obviously know, but you know, for those ignorant people who certainly aren't myself). And what a cycle is. And why are negative cycles hidden on purpose? Is that you're "run infinitely" setting? I don't understand why you'd want to hide that.

Reply:
Done. I had hidden the negative cycles because I didn't know anyone but me would be interested in running infinitely (though if someone looked at the code theyd see it called out in a comment).